android/app/src/main/java/io/peggyjo/peggymeter/HistoryTextFragment.java

```diff
@@ -31,6 +38,37 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
         dataController.addMoodListener(this);
         // Inflate the layout for this fragment
         View view = inflater.inflate(R.layout.fragment_history_text, container, false);
+        ListView table = view.findViewById(R.id.LOG_ENTRIES_CONTAINER);
+        table.setOnItemLongClickListener(new AdapterView.OnItemLongClickListener() {
+            @Override
+            public boolean onItemLongClick(AdapterView<?> adapterView, View view, int i, long l) {
+                LogEntry selectedEntry = (LogEntry)table.getItemAtPosition(i); //used instead of getSelectedItem because getSelectedItem causes a null pointer exception
+                EditText note = new EditText(getContext());
+
+                if(!selectedEntry.getComment().isEmpty()) {
+                    note.setText(selectedEntry.getComment());
+                }
+                AlertDialog.Builder builder = new AlertDialog.Builder(getContext());
+
+                if(!selectedEntry.getComment().isEmpty()) {
+                    builder.setTitle("Edit note");
+                } else {
+                    builder.setTitle("Add a note");
+                }
+                builder.setView(note);
+                AlertDialog editor = builder.create();
+                editor.setOnDismissListener(new DialogInterface.OnDismissListener() {
```

Comment:
This has been replaced with a button.

Reply:
Right.
What I mean is imagine you somehow long press on an item in the list view by mistake (you have no intention of editing the entry). Now a dialog pops up and you have only one option "Done" which will then invoke a Firebase write (an expensive operation because it involves a network write) regardless.
A rule of thumb in writing apps which involve database logic is to interact with your database only when really necessary.
This is why I'm asking you to (a) have a Cancel option and (b) check which button the user hit on the dialog window after it returns and if it was Cancel do all necessary cleanup without a database write.

Reply:
Right, I will do that.